File: c_third_lesson/src/service_client.cpp
#include "ros/ros.h"
#include "c_third_lesson/Service.h"

int main(int argc, char **argv) {
    ros::init(argc, argv, "service_client_node");

    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<c_third_lesson::Service>("add_two_ints");

    int32_t a = nh.param<int32_t>("/service_client_node/a", 0);
    int32_t b = nh.param<int32_t>("/service_client_node/b", 0);

    c_third_lesson::Service service;

    service.request.a = a;
    service.request.b = b;

    if(client.call(service)) {
        ROS_INFO("Sum %ld", (long int)service.response.sum);
    }else {
        ROS_ERROR("Failed to call service add_two_ints");
        return 1;
    }

    return 0;
}

File: c_third_lesson/src/service_client.py
#! /usr/bin/python3

import rospy
import sys
from c_third_lesson.srv import Service, ServiceRequest, ServiceResponse

def add(x: int, y: int):
    rospy.wait_for_service('add_two_ints')
    
    try:
        service = rospy.ServiceProxy('add_two_ints', Service)
        response = service(x, y)
        return response.sum
    except rospy.ServiceException as e:
        print("Service call failed: %s"%e)

def usage():
    return "%s [x y]"%sys.argv[0]


if __name__ == "__main__":
    if len(sys.argv) == 3:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
    else:
        print(usage())
        sys.exit(1)

    print("Requesting %s + %s"%(x, y))
    print("%s + %s = %s"%(x, y, add(x, y)))